### PR TITLE
[ADD] Support `CrossEntropyLoss` in KFAC

### LINKED ISSUE #44: Support `nn.CrossEntropyLoss` in KFAC

### PR DESCRIPTION
**Pull request by f-dangel**
Closes #44.

As a test, I am considering a data set consisting of a single sample. In that case, the Fisher-blocks are exactly Kronecker-factored, because there is no sum over multiple data points. In the limit of infinite MC-samples, KFAC thus approaches the block-diagonal Fisher.

### CODE REVIEW

File: curvlinops/kfac.py [hunk 1]
```diff
@@ -279,6 +280,23 @@ def draw_label(self, output: Tensor) -> Tensor:
...
+        elif isinstance(self._loss_func, CrossEntropyLoss):
+            # TODO For output.ndim > 2, the scale of the 'would-be' gradient resulting
+            # from these labels might be off
+            if output.ndim != 2:
+                raise NotImplementedError(
+                    "Only 2D output is supported for CrossEntropyLoss for now."
+                )
+            probs = output.softmax(dim=1)
+            # each row contains a vector describing a categorical
+            probs_as_mat = rearrange(probs, "n c ... -> (n ...) c")
```
Comment (runame) [inline curvlinops/kfac.py:293]: The `softmax` and `rearrange` operation could just be executed once before the for-loop over `range(self._mc_samples)` in a `_preprocess_model_output(output)` function or so.

**Review by runame — APPROVED** [review]
Besides the potential minor performance improvement in my comment, LGTM.

File: curvlinops/kfac.py (hunk 1 above)
Comment (f-dangel) [reply curvlinops/kfac.py:293]: Hm. Not sure if this will be relevant in practise, as I believe `mc_samples` is almost always `1`, at least for the use cases I know.
